File: Partake/app/in/partake/controller/api/debug/ErrorExceptionAPI.java
package in.partake.controller.api.debug;

import play.mvc.Result;
import in.partake.base.PartakeException;
import in.partake.controller.api.AbstractPartakeAPI;
import in.partake.model.dao.DAOException;

public class ErrorExceptionAPI extends AbstractPartakeAPI {

    public static Result get() throws DAOException, PartakeException {
        return new ErrorExceptionAPI().execute();
    }

    /**
     * RuntimeException が不意に起こった場合の対応をテスト
     * @return
     */
    @Override
    protected Result doExecute() throws DAOException {
        throw new RuntimeException("Some Error");
    }
}

File: app/controllers/api/debug/ErrorExceptionAPI.scala
package controllers.api.debug

import controllers.api.AbstractAPI
import controllers.ActionContext
import play.api.mvc.Request
import play.api.mvc.AnyContent
import play.api.mvc.PlainResult

/**
 * RuntimeException が不意に起こった場合の対応をテスト
 */
object ErrorExceptionAPI extends AbstractAPI[Unit, Unit] {
  override def parseRequest(request: Request[AnyContent])(implicit context: ActionContext): Unit = {
  }

  override def executeAction(params: Unit)(implicit context: ActionContext): Unit = {
  }

  override def renderResult(values: Unit)(implicit context: ActionContext): PlainResult = {
    throw new RuntimeException("Some Error");
  }
}
